### ISSUE TITLE
Include paths with '.' and '..' aliases should not be allowed

### ISSUE THREAD

**Issue by Mercotui** [cpplint/cpplint#432]
Hey,
I just noticed that this code passes cpplint 2.0.2:
```cpp
#include "../foo/bar.h"
```

I would expect this to be disallowed, as stated [here](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes):

> All of a project's header files should be listed as descendants of the project's source directory without use of UNIX directory aliases . (the current directory) or .. (the parent directory).

### Full test setup:
I have the file `foo/bar.cpp` with contents:
```cpp
// Copyright 2026 Acme Corp
#include "../include/foo/bar.h"

#include "../private/./test.h"
```
and the header `foo/bar.h` which is just an empty file, which needs to exist for the linter to run `CheckHeaderFileIncluded`.

I then run cpplint:
```bash
$ cpplint.py foo/bar.cpp
Done processing foo/bar.cpp
```

**Comment by Mercotui** [comment]
I see there was an opposite request a few years ago #139.
It seems that request resulted in some [special code](https://github.com/cpplint/cpplint/blob/develop/cpplint.py#L2757) to handle unix dir aliases, but this is only used to create a more verbose error message if a source file does not include its header, and one of the include paths uses unix dir aliases.

**Comment by Mercotui** [comment]
My motivation for wanting this check is: 
I'm usually very conscientious about splitting my code up into modular parts, and enforcing access to include directories with CMake. I don't want relative paths to bypass my CMake configuration.

But IDEs will often generate relative includes (i.e. when refactoring code).
I keep finding relative includes in my code, so I would really like cpplint to help me enforce this styleguide rule.

**Comment by cclauss** [comment]
There is a ruff rule for that so we should make that rule mandatory.
